pkg/network/ovn_kubernetes.go

```diff
@@ -0,0 +1,220 @@
+	data.Data["MTU"] = c.MTU
+
+	var ippools string
+	for _, net := range conf.ClusterNetworks {
+		if len(ippools) != 0 {
+			ippools += ","
+		}
+		ippools += fmt.Sprintf("%s/%d", net.CIDR, (32 - net.HostSubnetLength))
+	}
+	data.Data["OVN_cidr"] = ippools
+	data.Data["OVN_service_cidr"] = conf.ServiceNetwork
+
+	operCfg, err := controllerOVNConfig(conf)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to build controller config")
+	}
+	data.Data["NetworkControllerConfig"] = operCfg
+
+	nodeCfg, err := nodeOVNConfig(conf)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to build node config")
+	}
+	data.Data["NodeConfig"] = nodeCfg
+
+	manifests, err := render.RenderDir(filepath.Join(manifestDir, "network/ovn-kubernetes"), &data)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to render manifests")
+	}
+
+	objs = append(objs, manifests...)
+	return objs, nil
+}
+
+// validateOVNKubernetes checks that the ovn-kubernetes specific configuration
+// is basically sane.
+func validateOVNKubernetes(conf *netv1.NetworkConfigSpec) []error {
+	out := []error{}
+
+	if len(conf.ClusterNetworks) == 0 {
+		out = append(out, errors.Errorf("ClusterNetworks cannot be empty"))
+	}
+	oc := conf.DefaultNetwork.OVNKubernetesConfig
+	if oc != nil {
+		if oc.GenevePort != nil && (*oc.GenevePort < 1 || *oc.GenevePort > 65535) {
+			out = append(out, errors.Errorf("invalid GenevePort %d", *oc.GenevePort))
+		}
+
+		if oc.MTU != nil && (*oc.MTU < 576 || *oc.MTU > 65536) {
+			out = append(out, errors.Errorf("invalid MTU %d", *oc.MTU))
+		}
+	}
+
+	return out
+}
+
+// isOVNKubernetesChangeSafe currently returns an error if any changes are made.
+// In the future, we may support rolling out MTU or other alterations.
+func isOVNKubernetesChangeSafe(prev, next *netv1.NetworkConfigSpec) []error {
+	pn := prev.DefaultNetwork.OVNKubernetesConfig
+	nn := next.DefaultNetwork.OVNKubernetesConfig
+
+	if reflect.DeepEqual(pn, nn) {
+		return []error{}
+	}
+	return []error{errors.Errorf("cannot change ovn-kubernetes configuration")}
+}
+
+func fillOVNKubernetesDefaults(conf, previous *netv1.NetworkConfigSpec, hostMTU int) {
+	if conf.DefaultNetwork.OVNKubernetesConfig == nil {
+		conf.DefaultNetwork.OVNKubernetesConfig = &netv1.OVNKubernetesConfig{}
+	}
+
+	sc := conf.DefaultNetwork.OVNKubernetesConfig
+	if sc.GenevePort == nil {
+		var port uint32 = 6081
+		sc.GenevePort = &port
+	}
+	// MTU is currently the only field we pull from previous.
+	// If it's not supplied, we infer it from  the node on which we're running.
+	// However, this can never change, so we always prefer previous.
+	if sc.MTU == nil {
+		var mtu uint32 = uint32(hostMTU) - 100 // 100 byte geneve header
+		if previous != nil {
+			mtu = *previous.DefaultNetwork.OVNKubernetesConfig.MTU
+		}
+		sc.MTU = &mtu
+	}
+}
+
+func networkPluginName() string {
+	return "ovn-kubernetes"
+}
+
+//PHIL -- what is this?
+// controllerOVNConfig builds the contents of controller-config.yaml
```

Comment:
This creates the configuration file used by the openshift-sdn controller process. In the OVN daemonset, the config options will all come from environment variables whose values come from the ConfigMap, which is all set up in `renderOVNKubernetes` so there's no equivalent of this function. (Likewise with `nodeOVNConfig`.)

Reply:
yup, delete this function.